### PR TITLE
project: skeleton service, module and mocks

### PR DESCRIPTION
**Pull request by mikecutalo**
### Description

Open source the skeleton of the project api's, this will mostly be used for the mock server and serve as an example of how to implement the project api. The project module has a configuration option to override the default service that it depends on, which is necessary for any practical usage of this api, as you will need to implement the `{your company}.service.project` api in your own gateway.

Overriding the default service for the module can be done like so.
```yaml
  - name: clutch.module.project
    typed_config:
      "@type": types.google.com/clutch.config.module.project.v1.Config
      project_service_override: mycustomgateway.service.project
```

Example mock server response
```
➜  ~ curl --location --request POST 'http://localhost:8080/v1/project/getProjects' \
--header 'Content-Type: application/json' \
--data-raw '{
    "projects": [
        "clutch"
    ]
}'e | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   515  100   470  100    45  37603   3600 --:--:-- --:--:-- --:--:-- 85833
{
  "results": {
    "choiceapi": {
      "from": {
        "selected": true,
        "users": [
          "nom@meow.com"
        ]
      },
      "project": {
        "name": "choiceapi",
        "tier": "",
        "owners": [],
        "languages": [],
        "data": {},
        "dependencies": {
          "upstreams": {},
          "downstreams": {}
        },
        "oncall": null,
        "linkGroups": []
      }
    },
    "clutch": {
      "from": {
        "selected": true,
        "users": [
          "cat@meow.com"
        ]
      },
      "project": {
        "name": "clutch",
        "tier": "",
        "owners": [],
        "languages": [],
        "data": {},
        "dependencies": {
          "upstreams": {},
          "downstreams": {}
        },
        "oncall": null,
        "linkGroups": []
      }
    }
  },
  "partialFailures": []
}
```

### Testing Performed
unit / local

### CODE REVIEW

File: api/config/module/project/v1/project.proto [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+option go_package = "github.com/lyft/clutch/backend/api/config/module/project/v1;projectv1";
+
+import "validate/validate.proto";
+
+message Config {
+  // The default project service that clutch ships with is `clutch.service.project`
+  // The purpose of the open source project service is for the mock server and to serve
+  // as an boilerplate for private gateways to implement.
+  // Once your organization is ready to make use of the project API's in your private gateway,
```
Comment (danielhochman) [inline api/config/module/project/v1/project.proto:13]:
```suggestion
  // Once your organization is ready to make use of the project APIs in your private gateway,
```

File: api/config/module/project/v1/project.proto [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+package clutch.config.module.project.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/config/module/project/v1;projectv1";
+
+import "validate/validate.proto";
+
+message Config {
+  // The default project service that clutch ships with is `clutch.service.project`
+  // The purpose of the open source project service is for the mock server and to serve
+  // as an boilerplate for private gateways to implement.
```
Comment (danielhochman) [inline api/config/module/project/v1/project.proto:12]:
```suggestion
  // as a boilerplate for private gateways to implement.
```

File: backend/service/project/project.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	c := &client{
+		log:   logger,
+		scope: scope,
+	}
+
+	return c, nil
+}
+
+func (c *client) GetProjects(ctx context.Context, req *projectv1.GetProjectsRequest) (*projectv1.GetProjectsResponse, error) {
+	return nil, errors.New("not implemented")
```
Comment (danielhochman) [inline backend/service/project/project.go:36]: nit: use status.Error

File: backend/clutch-config.yaml [hunk 4]
```diff
@@ -35,7 +35,12 @@ modules:
   - name: clutch.module.envoytriage
   - name: clutch.module.k8s
   - name: clutch.module.kinesis
+  - name: clutch.module.project
+    typed_config:
+      "@type": types.google.com/clutch.config.module.project.v1.Config
+      project_service_override: clutch.service.project
```
Comment (danielhochman) [inline backend/clutch-config.yaml:41]: i've been thinking we could just add component aliasing to the proto a level up, but we can wait on that until the 1.18 refactor where we introduce contexts and use generics :)

so something like 
```yaml
  - name: clutch.module.project
    components:
      - name: clutch.service.project
        substitute: lyft.service.project
    typed_config:
      "@type": types.google.com/clutch.config.module.project.v1.Config
```

Comment (mikecutalo) [reply backend/clutch-config.yaml:41]: Oh yeah neat idea

**Review by smonero — APPROVED** [review]

**Pull request merged**